public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,318 @@
+include ../_util-fns
+
+:marked
+  Reactive forms allow explicit control over FormControls, FormArrays, and FormGroups.
+
+.l-main-section
+a#toc
+:marked
+  ## Contents
+
+  - [Setup](#setup)
+  - [Create a _reactive forms_ component](#create-comp)
+  - [Create the template](#create-template)
+  - [Import the _ReactiveFormsModule_](#import)
+  - [Update the app to use the form component](#update)
+  - [Intro to _FormBuilder_](#formbuilder)
+  - [Grouping _FormControls_](#grouping)
+  - [Inspecting form properties](#properties)
+  - [Setting data using _patchValue_ and _setValue_](#set-data)
+  - [Use _FormArray_ to add groups dynamically](#form-array)
+
+
+  Run the <live-example></live-example>.
+
+a#setup
+:marked
+  ## Setup
+  Create a new project folder (`reactive-forms`) and follow the steps in [QuickStart](../quickstart.html).
+
+include ../_quickstart_repo
+
+.l-main-section
+a#create-comp
+:marked
+  ## Create a _reactive forms_ component
+  The app needs a place for the form. Make a new file called `reactive-form.component.ts` in the `app` directory. Import `Component` from `@angular/core` and `AbstractControl`, `FormGroup`, and `FormControl` from `@angular/forms`:
+
++makeExample('reactive-forms/ts/app/reactive-form-v1.component.ts', 'reactive-comp-imports-v1','app/reactive-form.component.ts')(format=".")
+
+:marked
+  Here’s what each of them do:
+
+  - `Component`: Angular’s component decorator.
+
+  - `AbstractControl`: The base class for `FormArray`, `FormControl`, and `FormGroup`. It’s in charge of some of their shared behavior. You won’t have to instantiate it directly.
+
+  - `FormControl`: Allows you to instantiate individual **FormControls**. Also tracks the value and validity of an individual `FormControl`.
+
+  - `FormGroup`: Allows you to group **FormControls**, **FormArrays**, and other **FormGroups**.
+
+  Below the import statements, use the `@Component` decorator to provide metadata to Angular:
```

Comment:
Since this is decorating the class, consider using "above the class" instead of "below the import statements"